src/jdk.compiler/share/classes/com/sun/tools/javac/comp/DeferredAttr.java

```diff
@@ -1238,6 +1239,7 @@ public void visitReference(JCMemberReference tree) {
                     tree.getOverloadKind() != JCMemberReference.OverloadKind.UNOVERLOADED) {
                 stuckVars.addAll(freeArgVars);
                 depVars.addAll(inferenceContext.freeVarsIn(descType.getReturnType()));
+                depVars.addAll(inferenceContext.freeVarsIn(descType.getThrownTypes()));
```

Comment:
did the same for method references for completion but I couldn't find a test case that reproduced the issue with a method reference

Reply:
How about:
```
outer(nested(ConsiderExceptionTVarsInStuckExprs::mightThrow2));
...
static <C> void mightThrow2(C c) throws Exception {}
```

Reply:
yep that one makes the trick, thanks